apps/api/src/donations/events/stripe-payment.service.ts

```diff
@@ -110,6 +114,157 @@ export class StripePaymentService {
 
     const billingData = getPaymentData(paymentIntent)
 
-    await this.campaignService.donateToCampaign(campaign, billingData)
+    this.campaignService.donateToCampaign(campaign, billingData).then(() => {
+      this.checkForCompletedCampaign(metadata.campaignId as string)
+    })
+  }
+
+  @StripeWebhookHandler('customer.subscription.created')
+  async handleSubscriptionCreated(event: Stripe.Event) {
+    const subscription: Stripe.Subscription = event.data.object as Stripe.Subscription
+    const recurringDonation = await this.recurringDonationService.findSubscriptionByExtId(subscription.id)
+    if (recurringDonation) {
+      //we already have this subscription in our database
+      //stripe already sent us that event
+      return
+    }
+
+    Logger.log('[ handleSubscriptionCreated ]', subscription)
+
+    const metadata: DonationMetadata = subscription.metadata as DonationMetadata
+    if (!metadata.campaignId) {
+      throw new BadRequestException('Subscription metadata does not contain target campaignId. Subscription is: ' + subscription)
+    }
+
+    if (!metadata.personId) {
+      throw new BadRequestException('Subscription metadata does not contain target personId. Subscription is: ' + subscription)
+    }
+
+    const campaign = await this.campaignService.getCampaignByIdWithDefaultVault(metadata.campaignId)
+
+    const rdDto: CreateRecurringDonationDto = new CreateRecurringDonationDto()
+    rdDto.campaign = metadata.campaignId
+    rdDto.extSubscriptionId = subscription.id,
+    rdDto.extCustomerId = subscription.customer as string
+    rdDto.sourceVault = campaign["defaultVault"]
+
+    if (subscription.items.data.length == 0) {
+      throw new BadRequestException('Subscription does not contain any items. Subscription is: ' + subscription)
```

Comment:
We cast object `subscription` to string here. Maybe we need to pass only the subscription id

Reply:
fixed.